src/Sys/esp32Tools.cpp

```diff
@@ -0,0 +1,250 @@
+    case 17: //17 = Time Group1 reset CPU
+      return F("watchdog"); break;
+    case 18: //18 = super watchdog reset digital core and rtc module
+      return F("super watchdog"); break;
+
+    // misc
+    case 10: // 10 = Instrusion tested to reset CPU
+      return F("intrusion"); break;
+    case 19: //19 = glitch reset digital core and rtc module
+      return F("glitch"); break;
+    case 20: //20 = efuse reset digital core
+      return F("EFUSE reset"); break;
+    case 21: //21 = usb uart reset digital core
+      return F("USB UART reset"); break;
+    case 22: //22 = usb jtag reset digital core
+     return F("JTAG reset"); break;
+    case 23: //23 = power glitch reset digital core and rtc module
+      return F("power glitch"); break;
+
+    // unknown reason code
+    case 0:
+      return F(""); break;
+    default: 
+      return F("unknown"); break;
+  }
+}
+
+
+static esp_reset_reason_t getRestartReason() {
+  return(esp_reset_reason());
+}
+
+static String restartCode2InfoLong(esp_reset_reason_t reason) {
+    switch (reason) {
+      case ESP_RST_UNKNOWN:  return(F("Reset reason can not be determined")); break;
+      case ESP_RST_POWERON:  return(F("Restart due to power-on event")); break;
+      case ESP_RST_EXT:      return(F("Reset by external pin (not applicable for ESP32)")); break;
+      case ESP_RST_SW:       return(F("Software restart via esp_restart()")); break;
+      case ESP_RST_PANIC:    return(F("Software reset due to panic or unhandled exception (SW error)")); break;
+      case ESP_RST_INT_WDT:  return(F("Reset (software or hardware) due to interrupt watchdog")); break;
+      case ESP_RST_TASK_WDT: return(F("Reset due to task watchdog")); break;
+      case ESP_RST_WDT:      return(F("Reset due to other watchdogs")); break;
+      case ESP_RST_DEEPSLEEP:return(F("Restart after exiting deep sleep mode")); break;
+      case ESP_RST_BROWNOUT: return(F("Brownout Reset (software or hardware)")); break;
+      case ESP_RST_SDIO:     return(F("Reset over SDIO")); break;
+    }
+  return(F("unknown"));
+}
+
+static String restartCode2Info(esp_reset_reason_t reason) {
+    switch (reason) {
+      case ESP_RST_UNKNOWN:  return(F("unknown reason")); break;
+      case ESP_RST_POWERON:  return(F("power-on event")); break;
+      case ESP_RST_EXT:      return(F("external pin reset")); break;
+      case ESP_RST_SW:       return(F("SW restart by esp_restart()")); break;
+      case ESP_RST_PANIC:    return(F("SW error (panic or exception)")); break;
+      case ESP_RST_INT_WDT:  return(F("interrupt watchdog")); break;
+      case ESP_RST_TASK_WDT: return(F("task watchdog")); break;
+      case ESP_RST_WDT:      return(F("other watchdog")); break;
+      case ESP_RST_DEEPSLEEP:return(F("exit from deep sleep")); break;
+      case ESP_RST_BROWNOUT: return(F("Brownout Reset")); break;
+      case ESP_RST_SDIO:     return(F("Reset over SDIO")); break;
+    }
+  return(F("unknown"));
+}
+
```

Comment:
See also here: https://github.com/ewowi/StarMod/blob/19da7cb0145f5050920eeea2aab4618bbb0c9bc3/src/Sys/SysModSystem.cpp#L142

Duplicate code, one can be removed, which is the best to remove?

Reply:
Ah right, I'll try to remove the code in SysModSystem::addResetReasonsSelect.
I'm not much familiar with ArduinoJSON, but let's see.